go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go

```diff
@@ -703,14 +707,75 @@ func TestFkFuzzTest(t *testing.T) {
 	}
 }
 
-func validateReplication(t *testing.T) {
-	for _, keyspace := range clusterInstance.Keyspaces {
-		for _, shard := range keyspace.Shards {
-			for _, vttablet := range shard.Vttablets {
-				if vttablet.Type != "primary" {
-					checkReplicationHealthy(t, vttablet)
-				}
-			}
+// BenchmarkFkFuzz benchmarks the performance of Vitess unmanaged, Vitess managed and vanilla MySQL performance on a given set of queries generated by the fuzzer.
+func BenchmarkFkFuzz(b *testing.B) {
+	maxValForCol := 10
+	maxValForId := 10
+	insertShare := 50
+	deleteShare := 50
+	updateShare := 50
+	numQueries := 1000
+	// Wait for schema-tracking to be complete.
+	waitForSchemaTrackingForFkTables(b)
+	for i := 0; i < b.N; i++ {
+		// Clear out all the data to ensure we start with a clean slate.
+		startBenchmark(b)
+		// Create a fuzzer to generate and store a certain set of queries.
+		fz := newFuzzer(1, maxValForId, maxValForCol, insertShare, deleteShare, updateShare, SQLQueries, nil)
+		fz.noFkSetVar = true
+		var queries []string
+		for j := 0; j < numQueries; j++ {
+			genQueries := fz.generateQuery()
+			require.Len(b, genQueries, 1)
+			queries = append(queries, genQueries[0])
+		}
+
+		// Connect to MySQL and run all the queries
+		mysqlConn, err := mysql.Connect(context.Background(), &mysqlParams)
+		require.NoError(b, err)
+		// Connect to Vitess managed foreign keys keyspace
+		vtConn, err := mysql.Connect(context.Background(), &vtParams)
+		require.NoError(b, err)
+		utils.Exec(b, vtConn, fmt.Sprintf("use `%v`", unshardedKs))
+		// Connect to Vitess unmanaged foreign keys keyspace
+		vtUnmanagedConn, err := mysql.Connect(context.Background(), &vtParams)
+		require.NoError(b, err)
+		utils.Exec(b, vtUnmanagedConn, fmt.Sprintf("use `%v`", unshardedUnmanagedKs))
+
+		// First we make sure that running all the queries in both the Vitess modes and MySQL gives the same data.
+		// So we run all the queries and then check that the data in all of them matches.
+		runQueries(b, mysqlConn, queries)
+		runQueries(b, vtConn, queries)
+		runQueries(b, vtUnmanagedConn, queries)
+		for _, table := range fkTables {
+			query := fmt.Sprintf("SELECT * FROM %v ORDER BY id", table)
+			resVitessManaged, _ := vtConn.ExecuteFetch(query, 10000, true)
+			resMySQL, _ := mysqlConn.ExecuteFetch(query, 10000, true)
+			resVitessUnmanaged, _ := vtUnmanagedConn.ExecuteFetch(query, 10000, true)
+			require.True(b, compareResultRows(resVitessManaged, resMySQL), "Results for %v don't match\nVitess Managed\n%v\nMySQL\n%v", table, resVitessManaged, resMySQL)
+			require.True(b, compareResultRows(resVitessUnmanaged, resMySQL), "Results for %v don't match\nVitess Unmanaged\n%v\nMySQL\n%v", table, resVitessUnmanaged, resMySQL)
```

Comment:
Maybe it would be better to read if we move this to a different method and call it as  `benchmarkSetup` or something?